### PR TITLE
Show the sync as in progress on the settings page when the sync job is running

### PR DESCRIPTION
**Pull request by agamble**

### CODE REVIEW

File: app/components/settings/integration_component.rb [hunk 1]
```diff
@@ -12,4 +12,8 @@ class Settings::IntegrationComponent < ApplicationComponent
   def sentinel_connected_class
     "#{title.downcase.gsub(' ', '-')}-integration-connected"
   end
+
+  def sync_in_progress?
```
Comment (amisha29sh) [inline app/components/settings/integration_component.rb:16]: I think we need to scope by user here else this will fetch global latest in progress

Comment (agamble) [reply app/components/settings/integration_component.rb:16]: You are right!

Comment (agamble) [reply app/components/settings/integration_component.rb:16]: Done

File: app/models/sync_log.rb [hunk 2]
```diff
@@ -7,11 +7,27 @@ class SyncLog < ApplicationRecord
   validates :started_at, presence: true
   validates :user, presence: true
 
-  def self.start(task_name:, user:)
-    create!(task_name:, user:, started_at: Time.current)
+  def self.start!(task_name:)
+    raise StandardError, 'cannot start a task when one is already in progress' if latest(task_name)&.in_progress?
```
Comment (amisha29sh) [inline app/models/sync_log.rb:11]: Will need user here for the if condition too

Comment (agamble) [reply app/models/sync_log.rb:11]: I think this one is OK since we share the same scope as `start!`?

**Review by amisha29sh — APPROVED** [review]

**Pull request merged**